packages/web3/src/entitlement.ts

```diff
@@ -555,28 +555,35 @@ async function evaluateCheckOperation(
         operation.checkType !== CheckOperationType.NATIVE_COIN_BALANCE &&
         operation.contractAddress === zeroAddress
     ) {
-        throw new Error('Invalid contract address for check operation NATIVE_COIN_BALANCE')
+        throw new Error(
+            `Invalid contract address for check operation ${checkOpString(operation.checkType)}`,
+        )
     }
 
     if (
-        operation.checkType in
         [
```

Comment:
`in` keyword does not check for array contains in typescript, switched to `array.includes`